src/note.rs

```diff
@@ -169,14 +200,17 @@ impl Note {
     pub fn from_parts(
         recipient: Address,
         value: NoteValue,
+        asset: AssetBase,
         rho: Rho,
         rseed: RandomSeed,
     ) -> CtOption<Self> {
```

Comment:
_From str4d ([r2543506591](https://github.com/zcash/orchard/pull/471#discussion_r2543506591))_

```suggestion
    ) -> CtOption<Self> {
        Self::from_parts_internal(
            recipient,
            value,
            asset,
            rho,
            rseed,
            CtOption::new(rseed, 0u8.into()),
        )
    }

    /// Creates a `Note` from its component parts.
    ///
    /// This additionally permits constructing a [Split Input note], which is necessary
    /// for constructing certain patterns of bundles containing ZSA outputs. It is used by
    /// the PCZT code, which is the only place where these notes are serialized.
    ///
    /// Returns `None` if a valid [`NoteCommitment`] cannot be derived from the note.
    ///
    /// # Caveats
    ///
    /// See [`Self::from_parts`].
    ///
    /// [Split Input note]: https://zips.z.cash/zip-0226#split-notes
    pub(crate) fn from_parts_internal(
        recipient: Address,
        value: NoteValue,
        asset: AssetBase,
        rho: Rho,
        rseed: RandomSeed,
        rseed_split_note: CtOption<RandomSeed>,
    ) -> CtOption<Self> {
```

Reply:
FIXED